process.c

```diff
@@ -619,18 +619,30 @@ proc_s_last_status(VALUE mod)
     return rb_last_status_get();
 }
 
+static VALUE
+rb_process_status_initialize(VALUE self, VALUE pid, VALUE status, VALUE error)
```

Comment:
What's the semantics of `error`?
`status` is I guess the waitpid() wstatus?

Reply:
This is a little bit tricky problem.

The consumer of the result of `Process::Status.wait` can do the following:

- Return the status.
- Raise an error.

The fiber scheduler interface needs a way to transit this information without exceptions.

So, as well as the process `wstatus`, we need to capture `errno`. If you have a better idea about how this ca work, please feel free to suggest it. It's kind of tricky problem and it extends to things like `system`, backticks, etc. Eventually all process wait should go via scheduler (currently `system` does not because code is so tricky).